docs/developer/architecture/field-selectors.md

```diff
@@ -0,0 +1,131 @@
+		return []selector.SelectorName{"metadata.name", "metadata.alias"}
+	}
+	return nil
+}
+
+func (m *Device) ListSelectors() selector.SelectorNameSet {
+	return selector.NewSelectorFieldNameSet().Add("metadata.nameoralias")
+}
+```
+
+> [!NOTE]
+> - Mapping to multiple selectors will result in an OR condition between them.
+> - A mapped selector will be resolved first. It can override or hide an existing resolver defined by the resource model.
+
+### 3. Custom Selector Resolution
+
+For more advanced cases, selectors can be manually resolved.
+
+To enable this, the resource model must implement the following functions, which will be used by the field-selector during resolution:
+
+```go
+ResolveSelector(name selector.SelectorName) (*selector.SelectorField, error) 
+
+ListSelectors() selector.SelectorNameSet
+```
+
+#### Example Use Case: Whitelisting
+
+We use custom selectors to create a whitelist for the `Status` and `Spec` fields.
+For all Flight Control resources, `Status` and `Spec` fields are of type **JSONB**.
+Using explicit tagging on the resource model field results in allowing all **JSONB** keys to be queried.
+
+For example, tagging the `Status` field to allow it to be resolved:
+```go
+// The last reported state, stored as opaque JSON object.
+Status *JSONField[api.DeviceStatus] `gorm:"type:jsonb" selector:"metadata.status"`
+```
+
+The field-selector, in turn, resolves this field as type JSONB, automatically allowing keys like `status.updated.status` to be queried.
+
+
+While this approach is powerful, it introduces two challenges:
+
+- Type Constraints:
+Resolved keys only accept JSON values. For example:
+`status.updated.status="UpToDate"`
+
+- Loss of Control:
+Allowing all `Status` keys to be queried restricts our ability to whitelist and only support specific keys.
+
+#### Custom Selector for JSONB Fields
+
+Adding Custom Selector for `status.updated.status`
+```go
+func (m *Device) ResolveSelector(name selector.SelectorName) (*selector.SelectorField, error) {
+  if strings.EqualFold("status.updated.status", name.String()) {
+		return &selector.SelectorField{
+			Name:      name,
+			Type:      selector.String,
+			FieldName: "status.updated.status",
+			FieldType: "jsonb",
+		}, nil
+	}
+	return nil, nil
+}
+
+func (m *Device) ListSelectors() selector.SelectorNameSet {
+	return selector.NewSelectorFieldNameSet().Add("status.updated.status")
+}
+```
+
+> [!NOTE]
+> - Mapping a selector name using `MapSelectorName` to a custom selector is also supported. This can be useful for deprecating a key.
+> - When `FieldType` is defined as **JSONB**, the field-selector Adapts the `FieldName` to a valid **JSONB** key (e.g., `status -> 'updated' ->> 'status'`).
+> - If the `Type` is different from **JSONB**, the field-selector will cast the key to the corresponding type, and the selector will be processed as that type.
+> - A custom selector is resolved second. It can override or hide an existing resolver defined by the resource model.
```

Comment:
I think it would be best to point about the resolution order more explicitly.
```suggestion
> - A custom selector is resolved after mapped selectors. <Rest here>
```